--- origsrc/xforms-1.0.90/autogen.sh	2003-11-20 03:32:18.000000000 -0600
+++ src/xforms-1.0.90/autogen.sh	2007-08-28 23:55:20.859375000 -0500
@@ -4,18 +4,19 @@
 AUTOHEADER="autoheader"
 AUTOMAKE="automake -a -c --foreign"
 AUTOCONF="autoconf"
-ACINCLUDE_FILES="xformsinclude.m4 libtool.m4 cygwin.m4"
+LIBTOOLIZE="libtoolize -c -f"
+ACINCLUDE_FILES="xformsinclude.m4 cygwin.m4"
 
 # Discover what version of autoconf we are using.
 autoversion=`$AUTOCONF --version | head -n 1`
 
 echo "Using $autoversion"
 case $autoversion in
-    *2.5[234678])
+    *2.5[2-9]|*2.6[0-9])
 	;;
     *)
 	echo "This autoconf version is not supported by xforms."
-	echo "xforms only supports autoconf 2.5[234678]."
+	echo "xforms only supports autoconf >= 2.52."
 	exit
 	;;
 esac
@@ -27,6 +28,18 @@
 echo "done."
 
 # Generate the Makefiles and configure files
+if ( $LIBTOOLIZE --version ) < /dev/null > /dev/null 2>&1; then
+	echo "Installing libtool files..."
+	for dir in . ; do
+	    echo "        $dir"
+	    ( cd $dir ; $LIBTOOLIZE )
+	done
+	echo "done."
+else
+	echo "libtoolize not found -- aborting"
+	exit
+fi
+
 if ( $ACLOCAL --version ) < /dev/null > /dev/null 2>&1; then
 	echo "Building macros..."
 	for dir in . ; do
--- origsrc/xforms-1.0.90/config/cygwin.m4	2003-04-24 07:48:19.000000000 -0500
+++ src/xforms-1.0.90/config/cygwin.m4	2007-08-28 23:37:56.984375000 -0500
@@ -4,7 +4,7 @@
 AC_DEFUN(CHECK_WITH_CYGWIN,
 [
   case $host_os in
-  cygwin* | mingw* | pw32* )
+  mingw* | pw32* )
     if test "$GCC" = yes; then
       # Ensure MSVC-compatible struct packing convention.
       # Depends on GCC version. gcc2 uses -fnative-struct while
--- origsrc/xforms-1.0.90/configure.ac	2004-03-30 03:48:24.000000000 -0600
+++ src/xforms-1.0.90/configure.ac	2007-08-28 23:55:57.296875000 -0500
@@ -28,7 +28,6 @@
 AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 
 ### check for special systems
 AC_ISC_POSIX
--- origsrc/xforms-1.0.90/gl/Makefile.am	2003-11-20 03:33:11.000000000 -0600
+++ src/xforms-1.0.90/gl/Makefile.am	2007-08-28 23:37:56.984375000 -0500
@@ -7,7 +7,9 @@
 
 lib_LTLIBRARIES = libformsGL.la
 
-libformsGL_la_LDFLAGS = -version-info 1:0:0
+libformsGL_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+
+libformsGL_la_LIBADD = ../lib/libforms.la $(X_LIBS) $(X_PRE_LIBS) -lGL -lX11 $(X_EXTRA_LIBS)
 
 libformsGL_la_SOURCES = \
 	glcanvas.c \
--- origsrc/xforms-1.0.90/image/Makefile.am	2003-11-13 15:46:48.000000000 -0600
+++ src/xforms-1.0.90/image/Makefile.am	2007-08-28 23:37:57.000000000 -0500
@@ -7,7 +7,9 @@
 
 lib_LTLIBRARIES = libflimage.la
 
-libflimage_la_LDFLAGS = -version-info 1:0:0
+libflimage_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+
+libflimage_la_LIBADD = ../lib/libforms.la $(X_LIBS) $(X_PRE_LIBS) -lX11 $(X_EXTRA_LIBS) $(JPEG_LIB)
 
 libflimage_la_SOURCES = \
 	flimage.h \
--- origsrc/xforms-1.0.90/lib/Makefile.am	2003-11-20 03:33:16.000000000 -0600
+++ src/xforms-1.0.90/lib/Makefile.am	2007-08-28 23:37:57.000000000 -0500
@@ -17,7 +17,9 @@
 
 lib_LTLIBRARIES = libforms.la
 
-libforms_la_LDFLAGS = -version-info 1:0:0
+libforms_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+
+libforms_la_LIBADD = $(X_LIBS) $(X_PRE_LIBS) $(XPM_LIB) -lX11 $(X_EXTRA_LIBS)
 
 libforms_la_SOURCES = \
 	align.c \
